src/dc_mimeparser.rs

```diff
@@ -22,17 +23,18 @@ use crate::param::*;
 use crate::stock::StockMessage;
 use crate::types::*;
 use crate::x::*;
+use std::ptr;
 
 /* Parse MIME body; this is the text part of an IMF, see https://tools.ietf.org/html/rfc5322
 dc_mimeparser_t has no deep dependencies to Context or to the database
 (Context is used for logging only). */
 #[derive(Clone)]
 #[repr(C)]
 pub struct dc_mimepart_t {
-    pub type_0: libc::c_int,
+    pub type_0: Viewtype,
     pub is_meta: libc::c_int,
     pub int_mimetype: libc::c_int,
-    pub msg: *mut libc::c_char,
+    pub msg: Option<String>,
```

Comment:
Is `String` here safe?  I.e. is a mime part always valid UTF-8 rather than a vector of bytes?

Also, and I haven't looked deeply into this, but does `Option` help much here or add complexity over a message of length 0?

Reply:
@flub Previous version used `char *`, ergo it was null-terminated string, not bytevector. So I deduce that `msg` contains somehow transfer-encoded data, which is ascii-only. I agree, that `String` is too wide, but what else we have? `Vec<u8>` is type of raw bytes, not ascii string.

@dignifiedquire @r10 Opinions?

@flub Removing `Option` will make code in two places simplier and in on one place `src/dc_receive_imf.rs:630` it will change logic.

Reply:
I think going to `String` is good, all processing should happen on `String`s, with only very explicit entry and exit points, to ensure validity in as many places as possible.

Reply:
If there's a guarantee that this is ASCII than this is fine, but I wanted to make sure.

Sounds like removing the `Option` would be worth it IMHO.  But I'm fine not doing that in this PR to keep things smaller.